### DISCUSSION TITLE
Support for additional protocols

### DISCUSSION THREAD

**Discussion by oxeye-daniel** [projectdiscovery/nuclei#1119] [Ideas]
As for version 2.5.2 `Nuclei` supports a handful of protocols that are widely used on the internet. Having said that, there are plenty of protocols that aren't currently supported.

Since Nuclei is so widely used, it is a natural evolution to support more popular protocols, like:
- Message Queue protocols like `AMQP` and `Kafka`
- Networking protocols like `GRPC`, `WebSockets` and `HTTP2`
- Internet of Things (IoT) protocols like `MQTT`

**Comment by ehsandeep** [comment]
Few protocol supports that are already in development and related PR:

- [x] [Websocket](https://github.com/projectdiscovery/nuclei/pull/1066)
- [x] [SSL](https://github.com/projectdiscovery/nuclei/pull/1066)
- [ ] [HTTP2](https://github.com/projectdiscovery/rawhttp/pull/24)

**Comment by oxeye-daniel** [comment]
Here are a couple of examples and use cases for each protocol:

### `AMQP` / `Kafka` 
1. Probing open connections
2. Fuzzing/listing of available channels
3. Banner/version grabbing

### `GRPC`
1. Generic payload transmission
2. Validation of protobuf schemes

###  `MQTT`
1. Fuzzing/enumeration of available topics
2. Credential brute forcing
3. System information harvesting
